src/util/permissions.tsx

```diff
@@ -342,3 +367,43 @@ export const enablePermissionsFeature = (): boolean => {
 
   return hasAccessManagement && (hasOIDCSettings || userShowPermissions);
 };
+
+// each resource type has specific columns to display, which should uniquely identify the resource
+export const resourceOptionColumns: Record<string, (keyof ResourceDetail)[]> = {
+  image: ["aliases", "fingerprint", "description", "imageType"],
+  image_alias: ["name", "project"],
+  instance: ["name", "project"],
+  network: ["name", "project"],
+  network_acl: ["name", "project"],
+  network_zone: ["name", "project"],
+  profile: ["name", "project"],
+  storage_bucket: ["name", "project"],
+  storage_volume: ["name", "project", "pool"],
+  default: ["name"],
+};
+
+export const moveToFocusable = (
+  element: HTMLElement | null,
+  position: "prev" | "next",
+) => {
+  const focusableSelectors = [
+    'a:not([tabindex="-1"])',
+    'button:not([tabindex="-1"])',
+    'input:not([tabindex="-1"])',
+    'textarea:not([tabindex="-1"])',
+    'select:not([tabindex="-1"])',
+    '[tabindex]:not([tabindex="-1"])',
+  ].join(", ");
+
+  if (element) {
+    const focusableElements =
+      document.querySelectorAll<HTMLElement>(focusableSelectors);
+
+    // Find the index of the current element in the focusable elements list
+    // Then try to focus the next focusable element, if any
+    const index = Array.prototype.indexOf.call(focusableElements, element);
+    if (index > -1 && index < focusableElements.length - 1) {
+      focusableElements[index + (position === "next" ? 1 : -1)].focus();
+    }
+  }
+};
```

Comment:
the complexity comes from the tabbing interaction and I can't see another way of doing this since pressing tab while one of the select dropdown is open, we need to open the next dropdown. If we remove the tab to open functionality, then I agree with the above suggestion, we can keep things simple. 

Personally, I like the tabbing functionality as it makes the keyboard navigation a lot smoother for me. I think most of the complexity is in `moveToFocusable`, which is a general function that you would write once and not touch again. The logic in `handleTabbingForSelectors` seems also fairly straight forward no?

Reply:
@edlerd the changes as discussed have been implemented, I couldn't really implement a ref for the `add-entitlement` button as the `Button` component in react-component needs a ref input, which I think we can do outside the scope of this PR. Also made a few smaller adjustments/improvements:

1. When pressing `ESC` to close the select dropdown, keep focus on the select toggle.
2. After selecting a resource type, if there are no associated resources, then keep focus on the resource type selector.

Let me know if you agree with the above.